sys/kern/subr_bus.c

```diff
@@ -1226,7 +1231,7 @@ devclass_alloc_unit(devclass_t dc, device_t dev, int *unitp)
 	} else {
 		/* Unwired device, find the next available slot for it */
 		unit = 0;
-		for (unit = 0;; unit++) {
+		for (unit = 0; unit < DEVICE_UNIT_MAX; unit++) {
```

Comment:
INT_MAX

Reply:
`DEVICE_UNIT_MAX` could reasonably be set less than `INT_MAX`.  The most crucial factor is to ensure it doesn't overflow into the negative range, but I haven't yet seen any devices which had unit numbers in excess of 16 (which would be way too low as a setting, but `1<<31` is likely excessive).